auth.go

```diff
@@ -38,3 +38,38 @@ func (api *Client) SendAuthRevokeContext(ctx context.Context, token string) (*Au
 
 	return api.authRequest(ctx, "auth.revoke", values)
 }
+
+type listTeamsResponse struct {
+	Teams []Team `json:"teams"`
+	SlackResponse
+}
+
+type ListTeamsParameters struct {
+	Limit  int
+	Cursor string
+}
+
+// ListTeams returns all workspaces a token can access.
+func (api *Client) ListTeams(params ListTeamsParameters) ([]Team, *ListTeamsParameters, error) {
+	return api.ListTeamsContext(context.Background(), params)
+}
+
+// ListTeams returns all workspaces a token can access with a custom context.
+func (api *Client) ListTeamsContext(ctx context.Context, params ListTeamsParameters) ([]Team, *ListTeamsParameters, error) {
```

Comment:
[IMO]
I think the `ListTeamsParameter` is a struct for API calls, so I prefer to return next cursor instead of `ListTeamsParameter`. What do you think?

Reply:
do you mean returning `([]Team, string, error)`? I am okay with that too - I originally did this as I was following the example from https://github.com/stephenwan-opal/slack/blob/master/files.go#L288 and https://github.com/stephenwan-opal/slack/blob/master/remotefiles.go#L174

Let me know what you prefer

Reply:
Sorry for chiming in here; I'm neither the author nor maintainer, but I would very much like to use the work in this PR. 😄  I think that yes, returning `([]Team, string, error)` is what @kanata2 meant. I guess there are some inconsistencies in the SDK, but e.g. the `GetConversationsContext` method returns a string cursor: https://github.com/slack-go/slack/blob/657f4e78d9216ca7ccefeaf841b4b9623c32926b/conversation.go#L481

Reply:
i've updated the second commit to return the cursor instead of the entire params struct